src/lib.rs

```diff
@@ -1128,6 +1148,16 @@ impl EGraph {
     pub(crate) fn get_sort(&self, value: &Value) -> Option<&ArcSort> {
         self.proof_state.type_info.sorts.get(&value.tag)
     }
+
+    // Takes the last extract report and returns it, if the last command saved it.
+    pub fn take_extract_report(&mut self) -> Option<ExtractReport> {
```

Comment:
Why should these methods take the run report instead of returning a reference?

Reply:
It just seemed simpler and more efficient, but I can switch it to return a reference if you prefer.

Reply:
Yes, I think a reference would be better. It avoids mutation.

Reply:
Sounds good, fixed